### PR TITLE
Greedy few-shot sampling

### PR DESCRIPTION
**Pull request by shonohs**

### CODE REVIEW

File: vision_datasets/common/data_manifest.py [hunk 1]
```diff
@@ -324,6 +324,42 @@ def sample_subset_by_ratio(self, sampling_ratio):
...
+        Raises:
+            RuntimeError if it couldn't find n_shots images.
+        """
+        assert n_shots > 0
+        images = list(self.images)
+        rng = random.Random(random_seed)
+        rng.shuffle(images)
+
+        num_classes = len(self.labelmap) if not self.is_multitask else sum(len(x) for x in self.labelmap.values())
+        total_counter = collections.Counter({i: n_shots for i in range(num_classes)})
```
Comment (pj-ms) [inline vision_datasets/common/data_manifest.py]: This does not take negative images into consideration I suppose

File: vision_datasets/common/data_manifest.py [hunk 2]
```diff
@@ -324,6 +324,42 @@ def sample_subset_by_ratio(self, sampling_ratio):
...
+
+        Args:
+            n_shots (int): The number of images per class.
+            random_seed (int): Random seed to use.
+
+        Returns:
+            A samped dataset (DatasetManifest)
+
+        Raises:
+            RuntimeError if it couldn't find n_shots images.
```
Comment (pj-ms) [inline vision_datasets/common/data_manifest.py]: nit: 

RuntimeError if it couldn't find n_shots images for all classes

might be better.

File: vision_datasets/common/data_manifest.py [hunk 3]
```diff
@@ -324,6 +324,42 @@ def sample_subset_by_ratio(self, sampling_ratio):
         sampled_images = [self.images[i] for i in sampled_image_ids]
         return DatasetManifest(sampled_images, self.labelmap, self.data_type)
 
+    def sample_few_shots_subset_greedy(self, n_shots, random_seed=0):
+        """Greedy few-shots sampling method.
+        Randomly pick images from the original datasets until all classes have at least n_shots images.
```
Comment (pj-ms) [inline vision_datasets/common/data_manifest.py]: can we update the original sample_few_shots_subset docstring to emphasize that one is aiming for balanced sampling

One other thing is that the original few_shot method is greedy as well, I am thinking we can probably re-use the previous interface but having a flavor ['balanced', 'guaranteed'] or something, underlying functions can be two separated sampling method _few_shot_sampling_balanced, _few_shot_sampling_ganranteed or something

Comment (pj-ms) [reply vision_datasets/common/data_manifest.py]: Might as well mention the expected outcome of negative images as well

File: vision_datasets/common/data_manifest.py (hunk 1 above)
Comment (shonohs) [reply vision_datasets/common/data_manifest.py]: Right, added a comment about negative images.

File: vision_datasets/common/data_manifest.py (hunk 2 above)
Comment (shonohs) [reply vision_datasets/common/data_manifest.py]: fixed. thanks

File: vision_datasets/common/data_manifest.py (hunk 3 above)
Comment (shonohs) [reply vision_datasets/common/data_manifest.py]: I've updated the comments and mentioned negative images. Also, changed the parameter name from "n_shots" to "num_min_images_per_class" in order to emphasize the differences.

The two methods have different definition of the parameter, so I think it's safer to keep different interfaces.

**Pull request merged**